deploy/cephfs/kubernetes/v1.14+/csi-provisioner-rbac.yaml

```diff
@@ -58,19 +60,53 @@ metadata:
 subjects:
   - kind: ServiceAccount
     name: cephfs-csi-provisioner
+    # replace with non-default namespace name
     namespace: default
 roleRef:
   kind: ClusterRole
   name: cephfs-external-provisioner-runner
   apiGroup: rbac.authorization.k8s.io
 
+---
+apiVersion: policy/v1beta1
```

Comment:
do we need to create it by default?

Reply:
What do you mean?

Reply:
in helm charts this is optional, but here it not if this is not required by default. let's move this to a separate file. if it required user will create policy

Reply:
I tried hard to make this optional and find an easy way to test PodSecurityPolicies in e2e tests but couldn't.  

[PSP](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems) is an Admission Controller. It's not toggleable.

When it's provided as an admission plugin:
* All of the entities (e.g. user, service account) which can create a pod need to be associated with a PSP to be able to create said pod. Usually default privileged and restricted PSP's are created and associated with necessary entities when PSPs are enabled. This is why we have `scripts/psp.yaml` file which is provided in minikube [documentation](https://minikube.sigs.k8s.io/docs/tutorials/using_psp/).

When it's not provided as an admission plugin:
* PSP resources can be created and/or associated with an entity (user, service account). Since related admission controller is not active, they don't do anything. I tested this by adding PSP resources and not enabling PSP admission plugin on minikube with this [commit](https://github.com/ceph/ceph-csi/pull/785/commits/e5e67f3f509ea7e45514646ee2dc83cf1365dfcf) and [build](https://travis-ci.org/ceph/ceph-csi/builds/636923800?utm_medium=notification&utm_source=github_status).

IMHO it makes sense to include PSPs in e2e tests. This way we can find out whether a small change (eg. adding a new `hostPath`) is breaking the build or not. I tested this idea on this [commit](https://github.com/ceph/ceph-csi/pull/785/commits/5cb9e410fdd80fb5797e924752af3f2ce32529e8) and [build](https://travis-ci.org/ceph/ceph-csi/builds/636942110?utm_source=github_status&utm_medium=notification). However I also agree that this is a huge change. 

From top of my mind, I can think of below solutions:
1. Remove all PSP related changes from static kubernetes manifests that are used in e2e tests and remove PSP admission plugin flag from minikube initialization.

2. Keep all the changes as is. Make PSP admission plugin flag dynamic in `scripts/minikube.sh` script. Enable it for certain kubernetes versions in build matrix by passing an environment variable to some of the current builds or create new builds to test PSPs.

3. Move away PSP related changes from static kubernetes manifests to their own files. Add a new flag to `e2e/e2e_test.go` to apply these manifests dynamically. Make PSP admission plugin flag dynamic in `scripts/minikube.sh` script. Edit `/travis-functest.sh` so it's possible to call it an extra dynamic flag.

4. I'm open to your ideas.

Reply:
below is my suggestion

* Move  PSP related change in static templates to a new file
* Enable PSP by default on all kubernetes version( enabling on only one version may introduce a bug in other versions which are untested)
* Update E2E code to create PSP https://github.com/ceph/ceph-csi/blob/master/e2e/cephfs.go#L30 and https://github.com/ceph/ceph-csi/blob/master/e2e/rbd.go#L32

Reply:
>Minikube 1.5.2 with Kubernetes 1.16.x or higher supports PSP

minikube doc says this can be tested from kube 1.16.x onwards deployed by minikube let's enable it on kube 1.16.x onwards

Reply:
I also thought that is the case however when I enabled PSPs for all versions and removed my changes from static templates for version v1.13 in this [commit](https://github.com/ceph/ceph-csi/pull/785/commits/e628a2d2f9b73e7116f01782e079f17465843b77), tests failed as you can see in this [build](https://travis-ci.org/ceph/ceph-csi/builds/638026533). There might be a discrepancy between minikube documentation and actual implementation.

I'll still implement it as you suggested though.

Reply:
we need to check why it failed, but as we are planning to remove the support for v1.13 lets not worry about it  anymore

Reply:
```
STEP: checking provisioner statefulset/deployment is running
Jan 16 16:19:09.473: INFO: Waiting up to 10m0s for StatefulSet csi-cephfsplugin-provisioner to have all replicas ready
Jan 16 16:19:09.476: INFO: StatefulSet csi-cephfsplugin-provisioner found but there are 0 ready replicas and 1 total replicas.```

I've took this log snippet from that failed build. This is the expected behaviour when PSPs are enabled on a cluster and pod’s service account does not have any access to a permissive PSP via (Cluster)Role and (Cluster)RoleBinding.

Reply:
okay, then that should be fine, will rebase this PR once we remove kube 1.13 support from this repo